src/Ext/Techno/Body.Update.cpp

```diff
@@ -65,6 +65,70 @@ void TechnoExt::ExtData::ApplyInterceptor()
 	}
 }
 
+//test
+void TechnoExt::ExtData::DepletedAmmoActions()
+{
+	auto const pThis = this->OwnerObject();
+	auto const pType = pThis->GetTechnoType();
+	bool const isUnit = pThis->WhatAmI() == AbstractType::Unit;
+
+	if (!isUnit && (pType->Ammo <= 0))
+		return;
+
+	auto const pTypeExt = this->TypeExtData;
+	if (!pTypeExt->AmmoDepAction_AutoDeploy && !pTypeExt->AmmoDepAction_DeployBlock)
+		return;
+
+	auto const pUnit = abstract_cast<UnitClass*>(pThis);
+	bool const pThisCanDeploy = pUnit->Type->IsSimpleDeployer;
+
+	if (pThisCanDeploy)
+	{
+		if (pThis->Ammo <= 0)
+		{
+			if(pTypeExt->AmmoDepAction_AutoDeploy)
+			{
+				TechnoExt::UnitDeploySelf(pThis);
+				return;
+			}
+
+			if(pTypeExt->AmmoDepAction_DeployBlock)
+			{
+				TechnoExt::UnitDeployBlock(pThis);
+				return;
+			}
+		}
+	}
+	return;
```

Comment:
```suggestion
	if (!pThisCanDeploy)
	    return;

	if (pTypeExt->OnAmmoDepletion_AutoDeploy)
		TechnoExt::UnitDeploySelf(pThis);
	else if (pTypeExt->OnAmmoDepletion_DeployBlock)
		TechnoExt::UnitDeployBlock(pThis);
```

We already check for ammo at the beginning.

Reply:
At the beginning we check if Techno can have Ammo. Here I check exact ammo amount on unit and when it's 0 - I execute this stuff.

Reply:
Right, didn't notice one checks ammo for type and the other for instance.